nanoFramework.CoreInk/M5CoreInk.cs

```diff
@@ -0,0 +1,210 @@
+        private static GpioPin _led;
+        private static GpioButton _button;
+        private static GpioButton _left;
+        private static GpioButton _center;
+        private static GpioButton _right;
+        private static GpioButton _power;
+        private static GpioController _gpio;
+        private static Pcf8563 _rtc;
+
+        #region properties
+
+        /// <summary>
+        /// Gets the upper button.
+        /// </summary>
+        public static GpioButton RollerLeft
+        {
+            get
+            {
+                _left ??= new(37, GpioController, false, PinMode.Input);
+
+                return _left;
+            }
+        }
+
+        /// <summary>
+        /// Gets the upper button.
+        /// </summary>
+        public static GpioButton RollerRight
+        {
+            get
+            {
+                _right ??= new(39, GpioController, false, PinMode.Input);
+
+                return _right;
+            }
+        }
+
+        /// <summary>
+        /// Gets the upper button.
+        /// </summary>
+        public static GpioButton RollerButton
+        {
+            get
+            {
+                _center ??= new(38, GpioController, false, PinMode.Input);
+
+                return _center;
+            }
+        }
+
+        /// <summary>
+        /// Gets the upper button.
+        /// </summary>
+        public static GpioButton Button
+        {
+            get
+            {
+                _button ??= new(5, GpioController, false);
+
+                return _button;
+            }
+        }
+
+        /// <summary>
+        /// Gets the power button.
+        /// </summary>
+        public static GpioButton Power
+        {
+            get
+            {
+                _power ??= new(27, GpioController, false);
+
+                return _power;
+            }
+        }
+
+        /// <summary>
+        /// Gets the green led.
+        /// </summary>
+        public static GpioPin Led
+        {
+            get
+            {
+                _led ??= GpioController.OpenPin(10, PinMode.Output);
+
+                return _led;
+            }
+        }
+
+        /// <summary>
+        /// Gets the Buzzer.
+        /// </summary>
+        public static Buzzer Buzzer
+        {
+            get
+            {
+                // SetPinFunction already made in the static constructor
+                _buzzer ??= new(2);
+
+                return _buzzer;
+            }
+        }
+
+        /// <summary>
+        /// Gets the main <see cref="GpioController"/>.
+        /// </summary>
+        public static GpioController GpioController
+        {
+            get
+            {
+                _gpio ??= new();
+
+                return _gpio;
+            }
+        }
+
+        public static Pcf8563 RTC
+        {
+            get
+            {
+                _rtc ??= new(_device);
+
+                return _rtc;
+            }
+        }
+
```

Comment:
will you add the screen once it will be merged from the IoT Device repo?

Reply:
That's was the idea.

It is also possible to do a v1 without screen and later a v2 with it. But not sure if pertinent..

Reply:
There is no urgency, let's merge properly the screen and you'll be able to add it here. So it will be a nice version.